src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
Thanks for the feedback @eatdrinksleepcode it is a very insightful approach. I'm not sure how difficult it will be for that level of abstraction in the culture data but it would certainly be something we should consider, as I agree that both of the existing approaches aren't ideal. That said given the options of checking the OS version verse just asking the OS for the data I prefer the latter, if anything. I have similar concerns with tests that check the resource strings as they are very fragile in general and essentially hard coding the strings and expect them to never change or be localized. 

@stephentoub please decouple this issue from providing an API for getting the OS version. We have had that conversation numerous times but if you want to continue that discussion please file a separate issue explicitly for that.